File: python/ex_3.py
####################################################  
#
# 	@ Authors : SEKHRI Aymen
# 				MOHAMMED HACENE Tarek
#	
# 	@ Hint: you have to install all requirements
# 			from requirements.txt                    
#
#################################################### 

import numpy as np
import cv2 as cv
import matplotlib.pyplot as plt


# Load image
I = cv.imread("rose.jpeg")

# Extract height and width and channels
j, k, c = I.shape

# Scale factor
scale = 17

# Contruct new height and width based on new scale factor
x_new = round(j/scale)
y_new = round(k/scale)

print(x_new, y_new)
# Initialized a black image with the new scaled height and width
M = np.zeros((x_new, y_new, c), dtype="uint8")

# build the new scaled image
for ch in range(c): # loop through the channels
	for count1 in range(x_new): # loop through the new cols
		for count2 in range(y_new): # loop through the lines
			M[count1, count2, ch] = I[count1*scale, count2*scale, ch] # assigning to every pixel of the black image a new value


# plot the original image and the sampled image
cv.imshow("I Image", I)
cv.imshow('M Image', M)


cv.waitKey()
cv.destroyAllWindows()

File: matlab/ex_3.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
%   @ Authors : SEKHRI Aymen
%               MOHAMMED HACENE Tarek
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


clc 
close all
clear all

I = imread('rose.jpeg'); % load image 
[j, k,c] = size(I); % extract height , width , channles , from loaded image 

scale = 17; % scale factor

%x_new = round(j/scale); % new height based on original image height scaled with scale factor
%y_new = round(k/scale); % new width based on original image width scaled with scale factor

x_new = floor(j/scale); 
y_new = floor(k/scale); 

M = zeros(x_new,y_new,c); % initialize black image with the new height and width

for ch = 1:c % loop through every channel (R, G, B)
 for count1 = 1:x_new % loop through every col 
  for count2 = 1:y_new % loop through every row
    M(count1,count2,ch) = I(count1*scale,count2*scale,ch); % assign to every pixel of the black image a new value based on origial image pixel and scale factor 
  end
 end
end

subplot(1,2,1); imagesc(uint8(I)); axis tight; % plot origial image 
subplot(1,2,2); imagesc(uint8(M)); axis tight; % plot scaled image 